interview-questions.md

```diff
@@ -8,50 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: The difference between a parameter and an argument is that a parameter has a value that passes through an argument. 
+For example when writing a function through JavaScript
 
-Researched answer:
+const functionName = (parameter) => {
+    return parameter
+}
+
+console.log(functionName(argument))
+
+Researched answer: A parameter is a value that gets defined when a function gets invoked. As parameter also has an argument pass through the parameter during the invocation. An argument is usually an array like object that gets passed through the function.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: The JavaScript built in method .map() is used to iterate through the original array it is attached to using dot notation. The predefined parameters is logic that would change the elements in some way, while also keeping the original length of the array.
+For example:
+let array = [1, 2, 3]
+const changeArray = array.map((value) => {
+    return value * 2
+})
+
+console.log(changeArray)
+-> [2, 4, 6]
 
-Researched answer:
+Researched answer: The JavaScript built in method .map() calls a provided call back function which only invokes for arrays with assigned values. The call back function only takes in three arguments to the parameters: the values of the element, the index of the element, and the array object being mapped.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: The main difference between map and filter is that map keeps the original length of the array while filter only takes in the elements that fulfill its arguments(i.e. creates a new array with a smaller length than the original array)
 
-Researched answer:
+Researched answer: Map accesses the array and changes the values of each element, but it keeps the original length of the array. Filter uses boolean conditions in order to return a new array, if the element returns to true then it keeps the element, if it returns false then it excludes the element.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is when you go through each element of an array, or a character in a string, until a defined condition is met.
```

Comment:
A good way to expand this answer would be to talk about some of the different types of iteration you have used.